source/core/TDCodecVia.cpp

```diff
@@ -1834,33 +1835,33 @@ void TDViaFormatter::FormatIEEE754(TypeRef type, void* pData)
     Double value = ReadDoubleFromMemory(type, pData);
     Int32 len = 0;
     Boolean suppressInfNaN = _options._fmt.SuppressInfNaN();
+    Boolean quotedInfNaN = _options._fmt.QuotedNameInfNaN();
+
     if (isnan(value)) {
-#if 0
-        // TODO unit tests are getting different -NaNs in different cases.
-        if (signbit(value)) {
-            pBuff = "-nan";
-            len = 4;
-        } else {
-            pBuff = "nan";
-            len = 3;
-        }
-#else
         if (!suppressInfNaN) {
-            pBuff = _options._fmt.LongNameInfNaN() ?  "NaN" : "nan";
+            pBuff = _options._fmt.LongNameInfNaN() ?  "\"NaN\"" : "\"nan\"";
             len = 3;
+            if (quotedInfNaN)
```

Comment:
I'd be glad to add comments, but I'd rather not make it less efficient for a slight improvement in readability.